### PR TITLE
soc: nordic_nrf: nrf91: Add support for nRF9161 SiP / nRF9120 SoC

### PR DESCRIPTION
**Pull request by anangl**
The nRF9161 is technically a SiP (System-in-Package) that consists of the nRF9120 SoC and additional components like PMIC, FEM, and XTAL, so for nrfx/MDK the nRF9120 SoC is to be selected as the build target, but since the nRF9161 is what a user can actually see on a board, using only nRF9120 in the Zephyr build infrastructure might be confusing. That's why in the top level of SoC definitions (for user-configurable options in Kconfig, for example) the nRF9161 term is used and nRF9120 underneath.

This PR performs also some related clean-up and adds support for the nRF9161 DK.

### CODE REVIEW

File: boards/arm/nrf9161dk_nrf9161/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# nRF9161 DK NRF9161 board configuration
+
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_NRF9161DK_NRF9161 || BOARD_NRF9161DK_NRF9161_NS
+
+config BOARD
+	default "nrf9161dk_nrf9161"
+
```
Comment (joerchan) [inline boards/arm/nrf9161dk_nrf9161/Kconfig.defconfig:10]: Please add TF-M support

```
# By default, if we build for a Non-Secure version of the board,
# enable building with TF-M as the Secure Execution Environment.
config BUILD_WITH_TFM
       default y if BOARD_NRF9161DK_NRF9161_NS

if BUILD_WITH_TFM

# By default, if we build with TF-M, instruct build system to
# flash the combined TF-M (Secure) & Zephyr (Non Secure) image
config TFM_FLASH_MERGED_BINARY
       bool
       default y

endif # BUILD_WITH_TFM
```

**Review by gmarull — DISMISSED** [review]
lgtm, btw, is there a regulator driver for the embedded PMIC?

**Comment by anangl** [comment]
Rebased, updated hal_nordic SHA.

**Review by gmarull — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@anangl are you going to address the request to add tf-m support?

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**